### ISSUE TITLE
Support a subscription block operator

### ISSUE THREAD

**Issue by frmdstryr** [nucleic/enaml#347]
Would there be any interest in adding syntax for a "subscription block operator"? It would just be a shortcut which creates a declarative func and invokes it automatically when any of the parameters change.

For example something like:

```python
Label:
   text <= (a, b, c): # list of params to observe 
      if a:
         return 'format a'
      elif b and c:
         return 'format b'
      # etc..
```

Would be equivalent to this

```python
Label:
   func format_text(a, b, c):
      if a:
         return 'format a'
      elif b and c:
         return 'format b'
      # etc..
   text << format_text(a, b, c)
```

### Reasoning
When formatting text/values based on various changes I'm finding that I often either:
1. Use really ugly subscription operators using pythons inline `a if cond else b` statements because enaml automatically observes everything, but this makes code hard to follow and rather ugly.
2. Use a declarative func that takes all parameters I want to trigger an update. Much easier to read but it clutters code with temporary declarative funcs.

I'm unsure of how to handle cases when variables are renamed, perhaps something like default arguments ex `(a << obj.prop, b << obj.prop2, c=const)` 

Probably a stupid idea and may not be worth the effort but thought I'd at least ask :).

**Comment by sccolbert** [comment]
You don't need a declarative func if you pass it all the observable params.
Just use a module level function.

**Comment by frmdstryr** [comment]
When a function needs used in multiple places I do usually factor them out to module level utility functions.  I was mainly thinking of a way to reduce clutter/improve readability for single use subscriptions.

**Comment by frmdstryr** [comment]
After thinking about this more yesterday, I don't like it as suggested. There's ambiguity with the parameter names and invocation and it doesn't support the common case of a `None` check before accessing a member (which is the main reason for most of the inline `if else` statements in my code). 

Can enaml trace a block of code or only expressions?  

Ex Would using this
```python
Label:
   text << 
      if a: # Observe a
         return 'format a'
      elif b and c: # Observe b & c only if it accesses them
         return 'format b'
      # etc..
```
Instead of this

```python
Label:
   text << 'format a' if a else ('format b' if b and c else '')
```

be possible?  

Currently the later case works well already but it quickly get's hard to read if variable names are longer or more than one if statements is needed.

**Comment by sccolbert** [comment]
I think it will trace a declarative function.

**Comment by MatthieuDartiailh** [comment]
After discussing with @sccolbert I tried to achieve the same thing using declarative functions as illustrated in https://gist.github.com/MatthieuDartiailh/70c8f38dcc9266a0c200ab5e4618736f

However it appears that enaml does not automatically trace the body of the declarative function contrary to @sccolbert expectations. Should we change the behavior so that we automatically trace the body of declarative function ?
I still like @frmdstryr solution for 'simple' cases but I feel that tracing declarative functions could provide a broader solution. For example one could override the function in a derived class work, which is the function is used multiple times would avoid multiple editions ? Just thinking out loud here.

**Comment by frmdstryr** [comment]
Would they only trace when it's used in a `<<` operator?  What if I don't want it to re-trigger when certain members change or at all?  Would this make it too easy to introduce unexpected "noisy" updates? 

I'd be fine with this as long as there's a way to define a non traced version. Eg, a `def` vs `func` or add a `traced` keyword, `@observe` decorator or something.

Not that it matters but I've been using the change from the PR since it was opened without issues. I wouldn't mind having the ability to do both.

**Comment by frmdstryr** [comment]
The syntax in https://github.com/nucleic/enaml/issues/347#issuecomment-480267938 landed in https://github.com/nucleic/enaml/pull/348
